modules/sample.cpp

```diff
@@ -158,143 +158,192 @@ class CSampleMod : public CModule {
         return CONTINUE;
     }
 
-    void OnKick(const CNick& OpNick, const CString& sKickedNick, CChan& Channel,
-                const CString& sMessage) override {
+    void OnKickMessage(CKickMessage& Message) override {
+        const CNick& OpNick = Message.GetNick();
+        const CString sKickedNick = Message.GetKickedNick();
+        CChan& Channel = *Message.GetChan();
+        const CString sMessage = Message.GetReason();
         PutModule(t_f("{1} kicked {2} from {3} with the msg {4}")(
             OpNick.GetNick(), sKickedNick, Channel.GetName(), sMessage));
     }
 
-    void OnQuit(const CNick& Nick, const CString& sMessage,
-                const vector<CChan*>& vChans) override {
+    void OnQuitMessage(CQuitMessage& Message,
+                       const vector<CChan*>& vChans) override {
+        const CNick& Nick = Message.GetNick();
+        const CString sMessage = Message.GetReason();
+        VCString vsChans;
+        for (CChan* pChan : vChans) {
```

Comment:
This screams for C++20 `std::ranges::views::transform()`; but that can be improved later